src/main/java/frc/trigon/robot/utilities/CurrentWatcher.java

```diff
@@ -0,0 +1,54 @@
+package frc.trigon.robot.utilities;
+
+import edu.wpi.first.wpilibj.Notifier;
+import edu.wpi.first.wpilibj.Timer;
+
+import java.util.function.Supplier;
+
+/**
+ * A class that checks if the current past a certain limit for a certain amount of time.
+ */
+public class CurrentWatcher {
+    private double bellowCurrentTime;
+    private final Runnable runnable;
+    private final Supplier<Integer> currentSupplier;
+    private final int maxCurrent;
+    private final double timeThreshold;
+
+    /**
+     * Checks the current every 0.02 seconds, and it passes runs a runnable.
```

Comment:
Start this doc with something like "Construct a new CurrentWatcher that will check the current every 0.02 seconds."